\hypertarget{classerror__already__set}{}\section{error\+\_\+already\+\_\+set Class Reference}
\label{classerror__already__set}\index{error\_already\_set@{error\_already\_set}}


{\ttfamily \#include $<$pytypes.\+h$>$}

Inheritance diagram for error\+\_\+already\+\_\+set\+:\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[height=2.000000cm]{classerror__already__set}
\end{center}
\end{figure}
\subsection*{Public Member Functions}
\begin{DoxyCompactItemize}
\item 
\mbox{\hyperlink{classerror__already__set_a9f6456fb9f03460d943682037132e819}{error\+\_\+already\+\_\+set}} ()
\item 
\mbox{\hyperlink{classerror__already__set_a8bea74f30b56b414ee2629009cb252f4}{error\+\_\+already\+\_\+set}} (const \mbox{\hyperlink{classerror__already__set}{error\+\_\+already\+\_\+set}} \&)=default
\item 
\mbox{\hyperlink{classerror__already__set_a77db6b6c5770db3ec72f6e522375084a}{error\+\_\+already\+\_\+set}} (\mbox{\hyperlink{classerror__already__set}{error\+\_\+already\+\_\+set}} \&\&)=default
\item 
\mbox{\hyperlink{classerror__already__set_a18ad333137ef5b23916ff207a38e9a6a}{$\sim$error\+\_\+already\+\_\+set}} ()
\item 
\mbox{\hyperlink{_s_d_l__opengles2__gl2ext_8h_ae5d8fa23ad07c48bb609509eae494c95}{void}} \mbox{\hyperlink{classerror__already__set_a751b231e5c796447d867cf30007a7fc0}{restore}} ()
\item 
\mbox{\hyperlink{_s_d_l__opengles2__gl2ext_8h_ae5d8fa23ad07c48bb609509eae494c95}{void}} \mbox{\hyperlink{classerror__already__set_a471237e45e4049e9e9ec4bc049621eb1}{clear}} ()
\item 
\mbox{\hyperlink{asdl_8h_af6a258d8f3ee5206d682d799316314b1}{bool}} \mbox{\hyperlink{classerror__already__set_af5fb36e2230179ee533617b55ee90ce2}{matches}} (\mbox{\hyperlink{classhandle}{handle}} ex) const
\end{DoxyCompactItemize}


\subsection{Detailed Description}
Fetch and hold an error which was already set in Python. An instance of this is typically thrown to propagate python-\/side errors back through C++ which can either be caught manually or else falls back to the function dispatcher (which then raises the captured error back to python). 

\subsection{Constructor \& Destructor Documentation}
\mbox{\Hypertarget{classerror__already__set_a9f6456fb9f03460d943682037132e819}\label{classerror__already__set_a9f6456fb9f03460d943682037132e819}} 
\index{error\_already\_set@{error\_already\_set}!error\_already\_set@{error\_already\_set}}
\index{error\_already\_set@{error\_already\_set}!error\_already\_set@{error\_already\_set}}
\subsubsection{\texorpdfstring{error\_already\_set()}{error\_already\_set()}\hspace{0.1cm}{\footnotesize\ttfamily [1/3]}}
{\footnotesize\ttfamily error\+\_\+already\+\_\+set\+::error\+\_\+already\+\_\+set (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [inline]}}

Constructs a new exception from the current Python error indicator, if any. The current Python error indicator will be cleared. \mbox{\Hypertarget{classerror__already__set_a8bea74f30b56b414ee2629009cb252f4}\label{classerror__already__set_a8bea74f30b56b414ee2629009cb252f4}} 
\index{error\_already\_set@{error\_already\_set}!error\_already\_set@{error\_already\_set}}
\index{error\_already\_set@{error\_already\_set}!error\_already\_set@{error\_already\_set}}
\subsubsection{\texorpdfstring{error\_already\_set()}{error\_already\_set()}\hspace{0.1cm}{\footnotesize\ttfamily [2/3]}}
{\footnotesize\ttfamily error\+\_\+already\+\_\+set\+::error\+\_\+already\+\_\+set (\begin{DoxyParamCaption}\item[{const \mbox{\hyperlink{classerror__already__set}{error\+\_\+already\+\_\+set}} \&}]{ }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [default]}}

\mbox{\Hypertarget{classerror__already__set_a77db6b6c5770db3ec72f6e522375084a}\label{classerror__already__set_a77db6b6c5770db3ec72f6e522375084a}} 
\index{error\_already\_set@{error\_already\_set}!error\_already\_set@{error\_already\_set}}
\index{error\_already\_set@{error\_already\_set}!error\_already\_set@{error\_already\_set}}
\subsubsection{\texorpdfstring{error\_already\_set()}{error\_already\_set()}\hspace{0.1cm}{\footnotesize\ttfamily [3/3]}}
{\footnotesize\ttfamily error\+\_\+already\+\_\+set\+::error\+\_\+already\+\_\+set (\begin{DoxyParamCaption}\item[{\mbox{\hyperlink{classerror__already__set}{error\+\_\+already\+\_\+set}} \&\&}]{ }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [default]}}

\mbox{\Hypertarget{classerror__already__set_a18ad333137ef5b23916ff207a38e9a6a}\label{classerror__already__set_a18ad333137ef5b23916ff207a38e9a6a}} 
\index{error\_already\_set@{error\_already\_set}!````~error\_already\_set@{$\sim$error\_already\_set}}
\index{````~error\_already\_set@{$\sim$error\_already\_set}!error\_already\_set@{error\_already\_set}}
\subsubsection{\texorpdfstring{$\sim$error\_already\_set()}{~error\_already\_set()}}
{\footnotesize\ttfamily error\+\_\+already\+\_\+set\+::$\sim$error\+\_\+already\+\_\+set (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [inline]}}



\subsection{Member Function Documentation}
\mbox{\Hypertarget{classerror__already__set_a471237e45e4049e9e9ec4bc049621eb1}\label{classerror__already__set_a471237e45e4049e9e9ec4bc049621eb1}} 
\index{error\_already\_set@{error\_already\_set}!clear@{clear}}
\index{clear@{clear}!error\_already\_set@{error\_already\_set}}
\subsubsection{\texorpdfstring{clear()}{clear()}}
{\footnotesize\ttfamily \mbox{\hyperlink{_s_d_l__opengles2__gl2ext_8h_ae5d8fa23ad07c48bb609509eae494c95}{void}} error\+\_\+already\+\_\+set\+::clear (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [inline]}}

\mbox{\Hypertarget{classerror__already__set_af5fb36e2230179ee533617b55ee90ce2}\label{classerror__already__set_af5fb36e2230179ee533617b55ee90ce2}} 
\index{error\_already\_set@{error\_already\_set}!matches@{matches}}
\index{matches@{matches}!error\_already\_set@{error\_already\_set}}
\subsubsection{\texorpdfstring{matches()}{matches()}}
{\footnotesize\ttfamily \mbox{\hyperlink{asdl_8h_af6a258d8f3ee5206d682d799316314b1}{bool}} error\+\_\+already\+\_\+set\+::matches (\begin{DoxyParamCaption}\item[{\mbox{\hyperlink{classhandle}{handle}}}]{ex }\end{DoxyParamCaption}) const\hspace{0.3cm}{\ttfamily [inline]}}

Check if the currently trapped error type matches the given Python exception class (or a subclass thereof). May also be passed a tuple to search for any exception class matches in the given tuple. \mbox{\Hypertarget{classerror__already__set_a751b231e5c796447d867cf30007a7fc0}\label{classerror__already__set_a751b231e5c796447d867cf30007a7fc0}} 
\index{error\_already\_set@{error\_already\_set}!restore@{restore}}
\index{restore@{restore}!error\_already\_set@{error\_already\_set}}
\subsubsection{\texorpdfstring{restore()}{restore()}}
{\footnotesize\ttfamily \mbox{\hyperlink{_s_d_l__opengles2__gl2ext_8h_ae5d8fa23ad07c48bb609509eae494c95}{void}} error\+\_\+already\+\_\+set\+::restore (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [inline]}}

Give the currently-\/held error back to Python, if any. If there is currently a Python error already set it is cleared first. After this call, the current object no longer stores the error variables (but the {\ttfamily .what()} string is still available). 

The documentation for this class was generated from the following files\+:\begin{DoxyCompactItemize}
\item 
C\+:/\+Users/\+Riley Nolan/\+Desktop/\+School/\+Spring 2019/\+C\+S4850/git/finalproject-\/bluesentinel/\+Project/pybind11/include/pybind11/\mbox{\hyperlink{pytypes_8h}{pytypes.\+h}}\item 
C\+:/\+Users/\+Riley Nolan/\+Desktop/\+School/\+Spring 2019/\+C\+S4850/git/finalproject-\/bluesentinel/\+Project/pybind11/include/pybind11/\mbox{\hyperlink{pybind11_8h}{pybind11.\+h}}\end{DoxyCompactItemize}
